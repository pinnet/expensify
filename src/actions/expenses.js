import uuid from 'uuid';
export const addExpense = ({description = '',note = '',amount = 0,createdAt= new Date().getTime() }={}) => ({
    type: 'ADD_EXPENSE',
    expenses:{
        id: uuid(),
        description,
        amount,
        note,
        createdAt
    }  
});

export const removeExpense = ({id} = {}) => ({   
    type: 'REMOVE_EXPENSE',
    id
});

export const editExpense = (id,updates) => ({   
    type: 'EDIT_EXPENSE',
    id,
    updates
});

