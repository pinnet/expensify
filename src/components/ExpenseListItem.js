import React from 'react';
import moment from 'moment';
//import {connect} from 'react-redux';
import {Link} from 'react-router-dom';


const ExpenseListItem = ({id,description,amount,createdAt}) => (
 <div>
     <Link to={`/edit/${id}`}><h3>{description} : amount Â£{(amount / 100).toFixed(2)} : created {moment(createdAt).format('Do MMM YYYY')}</h3></Link>
 </div>
);
export default ExpenseListItem;