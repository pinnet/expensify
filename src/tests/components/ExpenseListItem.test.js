import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from './../../components/ExpenseListItem';
import expenses from './../fixtures/expenses';

it ('should render a fixture',() => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
});
