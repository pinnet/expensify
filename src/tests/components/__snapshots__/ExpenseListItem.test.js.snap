// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a fixture 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Gum
       : amount Â£
      1.95
       : created 
      1st Jan 1970
    </h3>
  </Link>
</div>
`;
